### ISSUE TITLE
getBufferedAmount always 0 with async loop

### ISSUE THREAD

**Issue by tadeuszwojcik** [uNetworking/uWebSockets.js#280]
Hi,
It must be something I don't understand regarding how it works in conjunction with Node.js event loop, but it's bugging me and I've spent last day or so trying to figure it out.

I've narrowed down my issue to this simple server that simply sends messages received from async generator as fast as possible. Problem is that in this scenario it never reports getBufferedAmount() > 0. It is able to send/receive ~200 000 messages/s. If I change the client to be slower (for example parsing received message 3 times) then getBufferedAmount() reports buffered data which is great, but  I receive ~800 000 messages/s which is bit unusual as slow client should cause lower throughput. I've tried EXPERIMENTAL_FASTCALL and cork, but without any difference. I must be missing something obvious here?

```js
const { App } = require('uWebSockets.js')
const wait = delayMS => new Promise(resolve => setTimeout(resolve, delayMS))

async function* asyncMessageGenerator() {
  while (true) {
    yield '{"hello":1}'
  }
}

App()
  .ws('/*', {
    open: async ws => {
      for await (const message of asyncMessageGenerator()) {
        const success = ws.send(message)
        if (!success) {
          while (ws.getBufferedAmount() > 0) {
            console.log('wait', new Date())
            await wait(1)
          }
        }
      }
    }
  })
  .listen(8080, () => {
    console.log('listening')
  })

```
and sample client:
```js
const { WebSocket } = require('ws')

var ws = new WebSocket('ws://localhost:8080/')
let count = 0
setInterval(() => {
  console.log('rps', count)
  count = 0
}, 1000)
ws.on('message', message => {
  //JSON.parse(message)
  //JSON.parse(message)
  //JSON.parse(message)
  count++
})
```

**Comment by uasan** [comment]
Your client code:
```JS
setInterval(() => {
  console.log('rps', count)
  count = 0
}
```
does not guarantee a time-honored call, if the microtask queue (ws.on) is busy, the timer may be called very late and your `RPS` does not correspond to the timer.

**Comment by tadeuszwojcik** [comment]
You're right, it's far from perfect, but it's not an issue in practice it has slight drift but it was just to show the difference

**Comment by uasan** [comment]
> You're right, it's far from perfect, but it's not an issue in practice, it has slight drift but it was just to show the difference

Ok, then maybe `ws.on`, called a handler 4x on one message

**Comment by tadeuszwojcik** [comment]
It doesn't seem to be the case either. Changing async data source to sync one (`while(true){}` loop for example) 'fixes' the issue - `getBufferedAmount` reports correctly

**Comment by alexhultman** [comment]
This is way too wide and deep of a subject to have a quick answer. I can't even decide where to start. I would need to run my own tests before knowing where to start, which I'm not going to.

What I can say is that your benchmark is unreliable and inaccurate.

**Comment by tadeuszwojcik** [comment]
Got it, thanks for the reply anyway, cheers.
